plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -270,6 +290,105 @@ private void testAddNestedColumnWithColumnMappingMode(String columnMappingMode)
                 .containsPattern("(delta\\.columnMapping\\.physicalName.*?){11}");
     }
 
+    @Test // regression test for https://github.com/trinodb/trino/issues/24121
+    void testPartitionValuesParsedCheckpoint()
+            throws Exception
+    {
+        for (ColumnMappingMode mode : List.of(ColumnMappingMode.ID, ColumnMappingMode.NAME, ColumnMappingMode.NONE)) {
+            testPartitionValuesParsedCheckpoint(mode, "boolean", ImmutableList.of("true", "false"), ImmutableList.of(true, false));
+            testPartitionValuesParsedCheckpoint(mode, "tinyint", ImmutableList.of("10", "20"), ImmutableList.of(Byte.valueOf("10"), Byte.valueOf("20")));
+            testPartitionValuesParsedCheckpoint(mode, "smallint", ImmutableList.of("100", "200"), ImmutableList.of((short) 100, (short) 200));
+            testPartitionValuesParsedCheckpoint(mode, "integer", ImmutableList.of("1000", "2000"), ImmutableList.of(1000, 2000));
+            testPartitionValuesParsedCheckpoint(mode, "bigint", ImmutableList.of("10000", "20000"), ImmutableList.of(10000L, 20000L));
+            testPartitionValuesParsedCheckpoint(mode, "real", ImmutableList.of("REAL '1.23'", "REAL '4.56'"), ImmutableList.of(1.23f, 4.56f));
+            testPartitionValuesParsedCheckpoint(mode, "double", ImmutableList.of("DOUBLE '12.34'", "DOUBLE '56.78'"), ImmutableList.of(12.34d, 56.78d));
+            testPartitionValuesParsedCheckpoint(
+                    mode,
+                    "decimal(3,2)",
+                    ImmutableList.of("0.12", "3.45"),
+                    ImmutableList.of(decimal("0.12", createDecimalType(3, 2)), decimal("3.45", createDecimalType(3, 2))));
+            testPartitionValuesParsedCheckpoint(mode, "varchar", ImmutableList.of("'alice'", "'bob'"), ImmutableList.of("alice", "bob"));
+            // TODO https://github.com/trinodb/trino/issues/24155 Cannot insert varbinary values into partitioned columns
```

Comment:
Do we have test with varbinary?

Reply:
Added `TestDeltaLakeConnectorTest#testInsertIntoUnsupportedVarbinaryPartitionType`.